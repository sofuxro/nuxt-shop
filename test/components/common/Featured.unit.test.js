import { factoryMount } from '~/test/utils/factories'
import flushPromises from 'flush-promises'

import Component from '~/components/common/Featured'

const mockFn = jest.fn((len) => [])
const store = {
    modules: {
        products: {
            namespaced: true,
            getters: {
                featuredProducts: (state) => mockFn
            }
        }
    }
}

let wrapper

beforeEach(() => {
    wrapper = factoryMount(Component, {
        store,
        stubs: {
            Product: { template: '<div />' }
        }
    })
})

afterEach(() => {
    wrapper.destroy()
})

describe('Featured', () => {
    test('vuetify breakpoints are working', async () => {
        // Snapshot
        expect(wrapper.element).toMatchSnapshot()

        wrapper.setData({ $vuetify: { breakpoint: { name: 'xl' } } })
        await flushPromises()
        expect(mockFn).toHaveBeenCalledWith(5)

        wrapper.setData({ $vuetify: { breakpoint: { name: 'lg' } } })
        await flushPromises()
        expect(mockFn).toHaveBeenCalledWith(4)

        wrapper.setData({ $vuetify: { breakpoint: { name: 'md' } } })
        await flushPromises()
        expect(mockFn).toHaveBeenCalledWith(3)
    })
})
