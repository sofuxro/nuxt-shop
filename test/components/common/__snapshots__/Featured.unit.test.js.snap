// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Featured vuetify breakpoints are working 1`] = `
<div>
  <h2>
    Featured Products 
    <span />
  </h2>
   
  <div
    class="product-list"
  />
</div>
`;
